src/aibtc-dao/extensions/payments-invoices/public/pay-invoice-by-resource-name.ts

```diff
@@ -67,6 +69,16 @@ async function main() {
     CONFIG.ACCOUNT_INDEX
   );
   const nextPossibleNonce = await getNextNonce(CONFIG.NETWORK, address);
+
+  // Add post-conditions to ensure sender has approved sufficient funds
+  // Note: The actual amount and token type should be read from the invoice
```

Comment:
We can do a read-only call to `get-resource-by-name` in the contract to get the value, related code:
https://github.com/aibtcdev/aibtcdev-contracts/blob/02470aa88060def8f11bbd8e05e78262eefe0f90/contracts/dao/extensions/aibtc-payments-invoices.clar#L345-L348